+sig/+node/Signal.m

```diff
@@ -285,28 +285,36 @@
       id.Node.DisplayInputs = this.Node.DisplayInputs;
     end
     
+    function fs = flatten(this)
+      % Returns a signal `fs` whose value is a "flattened" version of 
+      % `this`'s value. `this` is a subscriptable signal field whose value 
+      % is another signal. In other words, `flatten` "flattens" `this`'s 
+      % value - a signal - down to the value it holds.
+      
+      state = StructRef;
+      state.unappliedInputChanges = false;
+      fs = applyTransferFun(this, 'sig.transfer.flatten', state,...
+        '%s.flatten()');
+    end
+    
     function fs = flattenStruct(this)
       % Returns a signal `fs` whose value is a struct containing 
-      % "flattened" fields for the signal fields in the struct that 
-      % `this` has as its value. 
+      % "flattened" field values for the signal fields in `this`. In other
+      % words, `flattenStruct` "flattens" signal fields down to the values 
+      % they hold. (`this` can be a subscriptable signal, or a signal whose 
+      % value is a struct containing some signal fields). 
+      %
+      % `fs` uses `this` as a blueprint to wire up signals as inputs to
+      % itself. The values of the signal fields in `this` will be directly 
+      % set as `fs`'s struct field values. This is done in mexnet according
+      % to the transfer opCode.
       %
-      % Uses a signal (whose value is a struct with signal fields) as a 
-      % blueprint to wire up signals as inputs to this target. The values
-      % of these signal fields will be directly set as the target signal's
-      % struct field values. This is done in mexnet according to the 
-      % transfer opCode.
+      % Like `flatten`, but works on `this` directly, rather than `this`'s 
+      % signal fields.
       
       fs = applyTransferFun(this, 'sig.transfer.flattenStruct', [], '%s.flattenStruct()');
```

Comment:
We discussed adding back sig.transfer.flattenStruct or changing it.  Did you do this or ask Chris about it?

Reply:
Yeah didn't get a response from Chris. I can bug him about it again.